File: Python/45-扑克牌顺序.py
# -*- coding:utf-8 -*-
class Solution:
    def IsContinuous(self, numbers):
        # write code here
        if len(numbers) != 5:
            return False
        flag = 0
        MIN = 14
        MAX = -1
        for x in numbers:
            if x < 0 or x > 13:
                return False
            if x == 0:
                continue
            if (flag >> x) & 1 == 1:
                return False
            flag = flag or (1 << x)
            MIN = min(MIN, x)
            MAX = max(MAX, x)
            if MAX - MIN > 4:
                return False
        return True

File: C++/45-扑克牌顺序.cpp
class Solution
{
public:
    bool IsContinuous(vector<int> numbers)
    {
        if (numbers.size() != 5)
            return false;
        int min = 14, max = -1, flag = 0;
        for (int i = 0; i < numbers.size(); i++)
        {
            if (numbers[i] == 0)
                continue;
            if ((flag >> numbers[i]) & 1 == 1)
                return false;
            flag = flag | (1 << numbers[i]);
            min = numbers[i] < min ? numbers[i] : min;
            max = numbers[i] > min ? numbers[i] : max;
            if (max - min > 4)
                return false;
        }
        return true;
    }
};